src/main/java/org/folio/circulation/domain/representations/logs/CirculationCheckInCheckOutLogEventMapper.java

```diff
@@ -0,0 +1,140 @@
+import static org.folio.circulation.support.json.JsonPropertyWriter.write;
+
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import org.folio.circulation.domain.CheckInContext;
+import org.folio.circulation.domain.Loan;
+import org.folio.circulation.domain.LoanAndRelatedRecords;
+import org.folio.circulation.domain.UpdatedRequestPair;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class CirculationCheckInCheckOutLogEventMapper {
+  public static final String ITEM_SOURCE = "source";
+
+  private CirculationCheckInCheckOutLogEventMapper() {
+  }
+
+  /**
+   * This method returns {@link JsonObject} for check-in log event payload
+   * @param checkInContext check-in flow context {@link CheckInContext}
+   * @return check-in log event payload
+   */
+  public static JsonObject mapToCheckInLogEventJson(CheckInContext checkInContext) {
+    JsonObject logEventPayload = new JsonObject();
+
+    write(logEventPayload, LOG_EVENT_TYPE.value(), CHECK_IN.value());
+    write(logEventPayload, SERVICE_POINT_ID.value(), checkInContext.getCheckInServicePointId());
+
+    populateLoanData(checkInContext, logEventPayload);
+    populateItemData(checkInContext, logEventPayload);
+
+    ofNullable(checkInContext.getCheckInRequest())
+      .flatMap(checkInRequest -> ofNullable(checkInRequest.getClaimedReturnedResolution()))
+      .ifPresent(resolution -> write(logEventPayload, CLAIMED_RETURNED_RESOLUTION.value(), resolution.getValue()));
+
+    write(logEventPayload, REQUESTS.value(), getUpdatedRequests(checkInContext));
+
+    return logEventPayload;
+  }
+
+  /**
+   * This method returns {@link JsonObject} for check-out log event payload
+   * @param loanAndRelatedRecords check-out flow context {@link LoanAndRelatedRecords}
+   * @return check-out log event payload
+   */
+  public static JsonObject mapToCheckOutLogEventJson(LoanAndRelatedRecords loanAndRelatedRecords) {
+    JsonObject logEventPayload = new JsonObject();
+
+    write(logEventPayload, LOG_EVENT_TYPE.value(), CHECK_OUT.value());
+    write(logEventPayload, SERVICE_POINT_ID.value(), loanAndRelatedRecords.getLoan().getCheckoutServicePointId());
+
+    populateLoanData(loanAndRelatedRecords, logEventPayload);
+    populateItemData(loanAndRelatedRecords, logEventPayload);
+
+    write(logEventPayload, REQUESTS.value(), getUpdatedRequests(loanAndRelatedRecords));
+
+    return logEventPayload;
+  }
+
+  private static void populateItemData(CheckInContext checkInContext, JsonObject logEventPayload) {
+    ofNullable(checkInContext.getItem())
+      .ifPresent(item -> {
+        write(logEventPayload, ITEM_ID.value(), item.getItemId());
+        write(logEventPayload, ITEM_BARCODE.value(), item.getBarcode());
+        write(logEventPayload, ITEM_STATUS_NAME.value(), item.getStatusName());
+        write(logEventPayload, IS_CLAIMED_RETURNED.value(), item.isClaimedReturned());
```

Comment:
A claimed returned item has `Claimed returned` status. we can omit this flag, I think.